tested/languages/typescript/generators.py

```diff
@@ -0,0 +1,309 @@
+    # Handle basic types
+    value = as_basic_type(value)
+    if value.type in (BasicNumericTypes.INTEGER, BasicNumericTypes.REAL):
+        if not isinstance(value.data, SpecialNumbers):
+            return str(value.data)
+        elif value.data == SpecialNumbers.NOT_A_NUMBER:
+            return "NaN"
+        elif value.data == SpecialNumbers.POS_INFINITY:
+            return "Infinity"
+        else:
+            return "(-Infinity)"
+    elif value.type == BasicStringTypes.TEXT:
+        return json.dumps(value.data, ensure_ascii=False)
+    elif value.type == BasicBooleanTypes.BOOLEAN:
+        return str(value.data).lower()
+    elif value.type == BasicNothingTypes.NOTHING:
+        return "null"
+    elif value.type == BasicSequenceTypes.SEQUENCE:
+        assert isinstance(value, SequenceType)
+        return f"[{convert_arguments(value.data)}]"  # pyright: ignore
+    elif value.type == BasicSequenceTypes.SET:
+        assert isinstance(value, SequenceType)
+        return f"new Set([{convert_arguments(value.data)}])"  # pyright: ignore
+    elif value.type == BasicObjectTypes.MAP:
+        assert isinstance(value, ObjectType)
+        result = "new Map(["
+        for i, pair in enumerate(value.data):
+            result += "["
+            result += convert_statement(pair.key, True)
+            result += ", "
+            result += convert_statement(pair.value, True)
+            result += "]"
+            if i != len(value.data) - 1:
+                result += ", "
+        result += "])"
+        return result
+    elif value.type == BasicStringTypes.UNKNOWN:
+        assert isinstance(value, StringType)
+        return convert_unknown_type(value)
+    raise AssertionError(f"Invalid literal: {value!r}")
+
+
+def convert_function_call(call: FunctionCall, internal=False) -> str:
+    result = ""
+    if not internal:
+        result += "await "
+    if call.type == FunctionType.CONSTRUCTOR:
+        result += "new "
+    if call.namespace:
+        result += convert_statement(call.namespace, True) + "."
+    result += call.name
+    if call.type != FunctionType.PROPERTY:
+        result += f"({convert_arguments(call.arguments)})"  # pyright: ignore
+    return result
+
+
+def convert_statement(statement: Statement, internal=False, full=False) -> str:
+    if isinstance(statement, Identifier):
+        return statement
+    elif isinstance(statement, FunctionCall):
+        return convert_function_call(statement, internal)
+    elif isinstance(statement, Value):
+        return convert_value(statement)
+    elif isinstance(statement, PropertyAssignment):
+        return (
+            f"{convert_statement(statement.property, True)} = "
+            f"{convert_statement(statement.expression, True)}"
+        )
+    elif isinstance(statement, VariableAssignment):
+        if full:
+            prefix = "let "
+        else:
+            prefix = ""
+        return (
+            f"{prefix}{statement.variable} = "
```

Comment:
Interesting that the typescript judge does not type it's variables. (I think this is the place where types should be added)

I would not add it in this pr, as it is working as is and adding types can be a hassle. But I would assume as a user that the statements rendered by the judge would be typed.
So it might be relevant to create an issue to remember this for the future and add it in a future pr

Reply:
That's a very good point. This did not cross my mind when implementing this. I don't think this would be as easy either. However Java or C could probably help me out in that sector. I'll make an issue about this and will try to get to this when I've got the time.